src/tree/ResourceTreeDataProviderBase.ts

```diff
@@ -90,7 +90,7 @@ export abstract class ResourceTreeDataProviderBase extends vscode.Disposable imp
                 if (child instanceof BranchDataItemWrapper) {
                     return child;
                 }
-                return this.state.wrapItemInStateHandling(child, (item) => this.notifyTreeDataChanged(item));
```

Comment:
I'm going to refactor and rename this method so it makes more sense when it should be used vs. when the emitter should be used directly.

The gist is that `notifyTreeDataChanged` is for branch items. We need to rename it so that is easily understood from the code.